hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -931,6 +932,7 @@ private void startCommit(String instantTime, String actionType, HoodieTableMetaC
         "Found pending restore in active timeline. Please complete the restore fully before proceeding. As of now, "
             + "table could be in an inconsistent state. Pending restores: " + Arrays.toString(inflightRestoreTimeline.getInstantsAsStream()
             .map(instant -> instant.getTimestamp()).collect(Collectors.toList()).toArray()));
+    TimestampUtils.validateForLatestTimestamp(metaClient, instantTime);
```

Comment:
As discussed, this will need to be under a transaction as well.

Reply:
addressed this